katdal/test/test_lazy_indexer.py

```diff
@@ -73,8 +73,64 @@ def test_too_many_axes(self):
         self._test_index_error(np.s_[0, 0, 0, 0])
 
 
+class TestDaskGetitem(object):
+    """Test the :func:`~katdal.lazy_indexer._dask_getitem` function."""
+    def setup(self):
+        # Derived from example in NEP 21
+        shape = (5, 6, 7, 8)
+        self.data = np.arange(np.product(shape)).reshape(shape)
+        self.data_dask = da.from_array(self.data, chunks=(1, 2, 1, 2))
+
+    def _check_shape(self, indices, shape):
+        us = _dask_getitem(self.data_dask, indices).compute()
+        np.testing.assert_equal(us.shape, shape)
+
+    def _compare_indexing(self, indices):
+        npy = self.data[indices]
+        us = _dask_getitem(self.data_dask, indices).compute()
+        np.testing.assert_array_equal(us, npy)
+
+    def _test_with(self, indices, expected_error=None):
+        if expected_error is None:
+            self._compare_indexing(indices)
+        else:
+            with assert_raises(expected_error):
+                self._compare_indexing(indices)
+
+    def test_legacy_fancy_indexing(self):
+        self._test_with(np.s_[[0], ...])
+        self._test_with(np.s_[:, [0], ...])
+        self._test_with(np.s_[:, [0], [0], :], AssertionError)  # future error
+        self._test_with(np.s_[:, [0], :, [0]], AssertionError)  # future error
+        self._test_with(np.s_[:, [0], 0, :])
+        self._test_with(np.s_[:, [0], :, 0], AssertionError)  # future error
+        # Create boolean index with one True value for the last two dimensions
+        bindx = np.zeros((7, 8), dtype=np.bool_)
+        bindx[0, 0] = True
+        self._test_with(np.s_[:, 0, bindx], IndexError)  # dask can't handle it
+        self._test_with(np.s_[0, :, bindx], IndexError)  # dask can't handle it
+        self._test_with(np.s_[[0], :, bindx], IndexError)  # dask failure
+        self._test_with(np.s_[:, [0, 1], bindx], IndexError)  # dask failure
+
+    def test_outer_indexing(self):
+        self._check_shape(np.s_[:, [0], [0, 1], :], (5, 1, 2, 8))
+        self._check_shape(np.s_[:, [0], :, [0, 1]], (5, 1, 7, 2))
+        self._check_shape(np.s_[:, [0], 0, :], (5, 1, 8))
+        self._check_shape(np.s_[:, [0], :, 0], (5, 1, 7))
+        # Restrict ourselves to more practical 1-D boolean indices
+        bindx2 = np.zeros(7, dtype=np.bool_)
+        bindx2[0] = True
+        bindx3 = np.zeros(8, dtype=np.bool_)
+        bindx3[0] = True
+        bindx3[3] = True
+        self._check_shape(np.s_[:, 0, bindx2, bindx3], (5, 1, 2))
+        self._check_shape(np.s_[0, :, bindx2, bindx3], (6, 1, 2))
+        self._check_shape(np.s_[[0], :, bindx2, bindx3], (1, 6, 1, 2))
+        self._check_shape(np.s_[:, [0, 1], bindx2, bindx3], (5, 2, 1, 2))
+
```

Comment:
Since bindx2 will be simplified, and `[0, 1]` could potentially be simplified (even if it isn't yet), in future these tests might not actually cover the oindexing fallback path at all.

Also add this test: it currently fails because the oindexing code doesn't correctly account for scalar indices:
```
self._check_shape(np.s_[:, 0, [0, 2], [1, 3, 5]], (5, 2, 3))
```
I think some tests with `np.newaxis` and `...` are needed too.

Reply:
Isn't this test only testing the fast path? I think all those those fancy indices will get turned into slices by `_simplify_index`. The same applies to many of the other tests now that you're simplifying strided accesses.